src/main/java/com/google/ehub/servlets/ProfileServlet.java

```diff
@@ -0,0 +1,40 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns HTML that contains the user's profile information.**/
+@WebServlet("/profile-data")
+public class ProfileServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.sendRedirect("/ProfilePage.html");
+  }
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String user_email = userService.getCurrentUser().getEmail();
+    UserProfile newUser = UserDataManager.getUserProfile(user_email);
+    response.setContentType("application/json");
+    response.getWriter().println(convertToJson(newUser));
+  }
+  private String convertToJson(UserProfile profile) {
+    Gson gson = new Gson();
+    return gson.toJson(profile);
+  }
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
The same method getParameter() seems to be implemented twice in ProfileServlet and LoginServlet, might want to consider making a Utils.java class with this methods so that we can all re-use them.

Reply:
How do I do that?

Reply:
Create a new class Utils.java and add that method there and make it public and static. Then call this method in the classes that you are using it.